doc/source/matching_strategies.rst

```diff
@@ -0,0 +1,172 @@
+
+Grid-Free Matrix
+----------------
+
+Grid-fees are calculated for each order using the following grid fee matrix:
+
+.. math::
+
+    \[\left[ {\begin{array}{cc} 0 & 1 \\ 1 & 0 \\ \end{array} } \right]\]
+
+Bids and offers
+---------------
+
+Bids are placed by actors looking to buy energy and offers are placed by actors looking to sell
+energy. For this example, using the network depicted above, 5 orders consisting of 3 bids and 2 asks are
+added into the market.
+
+**The Bids**
+
+Two bids, each for 0.1 energy have been placed by Actor 1 in Cluster 1, one bid is set at the price
+of 10 and another bid at the price of 7. The third bid is placed by Actor 0 in Cluster 0 and is
+for 0.1 energy at the price 10.
+
+**The Offers**
+
+Both offers added into the market are placed by Actor 3 in Cluster 1. Both are for 0.1 energy
+with one set at the price of 6 and another at the price of 4.
+
+Below shows the syntax for inputting the bids and offers:
+
+.. code:: python
+
+    Order = namedtuple("Order", ("type", "time", "actor_id", "cluster", "energy", "price"))
+
+    # Add bids
+    m.accept_order(Order(-1, 0, 1, 1, 0.1, 10))
+    m.accept_order(Order(-1, 0, 1, 1, 0.1, 7))
+    m.accept_order(Order(-1, 0, 0, 0, 0.1, 10))
+
+    # Add asks
+    m.accept_order(Order(1, 0, 3, 1, 0.1, 6))
+    m.accept_order(Order(1, 0, 3, 1, 0.1, 4))
+
+Pay-as-Bid Algorithm
+====================
+
+Using the Pay-as-Bid algorithm (found here: :ref:`market`), buyers are able to place bids in the
+market, alongside
+the
+asks placed by sellers. At each timestamp, bids are sorted in descending order and asks
+are sorted in ascending order. Each offer is then compared with each bid, if the bid is greater
+than the ask, the ask and the bid are matched.
+
+The table below illustrates the matching of the :ref:`example_scenario`:
+
+
+**Final Matches (Output):**
+
++--------------------------------+-------------------------------+----------------+
+| Bids                           | Asks                          | Matched Price  |
++================================+===============================+================+
+| actor 0, order_id 2, price 10  | actor 3, order_id 4, price 4  | 10             |
++--------------------------------+-------------------------------+----------------+
+| actor 1, order_id 0, price 10  | actor 3, order_id 3, price 6  | 10             |
++--------------------------------+-------------------------------+----------------+
+| actor 1, order_id 1, price 7   |                               |                |
++--------------------------------+-------------------------------+----------------+
+
+Two-sided Pay-as-Clear Algorithm
+================================
+
+Similarly to Pay-as-Bid, at each timestamp, Two-sided Pay-as-Clear (found here: :ref:`2pac`)
+sorts bids sorts bids in descending order and asks in ascending order. After the matching has
+taken place at this timestep, the highest ask price to be matched, known as the *clearing price*,
```

Comment:
... time slot, highest matched ask price becomes the *clearing price* for all matched orders of this time slot.